x/simulation/simulate.go

```diff
@@ -230,6 +231,12 @@ func SimulateFromSeed(
 		logWriter.AddEntry(EndBlockEntry(blockHeight))
 
 		if config.Commit {
+			for _, endBlockFn := range endBlockFns {
+				if err := endBlockFn(ctx); err != nil {
+					return true, params, fmt.Errorf("endBlockFn failed at height %d: %w", blockHeight, err)
+				}
+			}
+			app.SimWriteState()
```

Comment:
```
res, err = app.internalFinalizeBlock(context.Background(), req)
	if res != nil {
		res.AppHash = app.workingHash()
	}
```

this is what normal execution do, maybe you can just add a `workingHash` call here, which will do the `app.finalizeBlockState.ms.Write()`.